File: Abstractions/Types/SettingsModel.cs
namespace Timeoff.Types
{
    public abstract record SettingsModel
    {
        public string Name { get; init; }

        public string Country { get; init; }

        public string DateFormat { get; init; }

        public string TimeZone { get; init; }

        public int CarryOver { get; init; }

        public bool ShowHoliday { get; init; }

        public bool HideTeamView { get; init; }
    }
}

File: timeoff.ui/src/app/services/company/settings.model.ts
import { ScheduleModel } from '@models/schedule.model';
import { LeaveTypeModel } from './leave-type.model';

export interface SettingsModel {
    name: string;
    country: string;
    dateFormat: string;
    timeZone: string;
    carryOver: number;
    showHoliday: boolean;
    hideTeamView: boolean;
    leaveTypes: LeaveTypeModel[];
    schedule: ScheduleModel;
}
